docs/creating-components-1.md

```diff
@@ -176,67 +166,44 @@ module MathModule {
 
 }
 ```
-> Note: this is an old way of formatting an fpp file but the content is entirely correct.
-
 
 ## About this Component 
  
-The above code defines much of the autocoded implementation of  `MathSender`. Generally speaking, while the user defines the types of inputs and outputs, the autocoders handle how the inputs and outputs are dealt with. `MathSender`is **active**, which means it has its
-own thread.
-
-Inside the definition of the `MathSender` component are
-several specifiers. We have divided the specifiers into five groups. Modern F' will not divide ports into groups like this, however the types of ports are still the same: @TODO (read this)
-
-1. **General ports:** These are user-defined ports for
-application-specific functions.
-There are two general ports: an output port `mathOpOut`
-of type `MathOp` and an input port `mathResultIn` of
-type `MathResult`.
-Notice that these port specifiers use the ports that
-you defined.
-The input port is **asynchronous**.
-This means that invoking the port (i.e., sending
-data on the port) puts a message on a queue.
-The handler runs later, on the thread of this component.
-
-2. **Special ports:** These are ports that have a special
-meaning in F Prime.
-There are ports for registering commands with the dispatcher,
-receiving commands, sending command responses, emitting
-event reports, emitting telemetry, and getting the time.
-
-3. **Commands:** These are commands sent from the ground
-or from a sequencer and dispatched to this component.
+The above code defines `MathSender` component. The component is **active**, which means it has its own thread.
+
+Inside the definition of the `MathSender` component are several specifiers. We have divided the specifiers into five groups.
+
+1. **General ports:** These are user-defined ports for application-specific functions.
+There are two general ports: an output port `mathOpOut` of type `OpRequest` and an input port `mathResultIn` of type `MathResult`.
+Notice that these port specifiers use the ports that you defined.
+The input port is **asynchronous**. This means that invoking the port (i.e., sending data on the port) puts a message on a queue. The handler runs later, on the thread of this component.
+
+2. **Special ports:** These are ports that have a special meaning in F Prime.
+There are ports for registering commands with the dispatcher, receiving commands, sending command responses, emitting event reports, emitting telemetry, and getting the time.
+
+3. **Commands:** These are commands sent from the ground or from a sequencer and dispatched to this component.
 There is one command `DO_MATH` for doing a math operation.
```

Comment:
`DO_MATH` is an asynchronous command used to perform a math operation. When an asynchronous command arrives,...